src/util/simplify_expr_pointer.cpp

```diff
@@ -359,8 +359,8 @@ simplify_exprt::simplify_pointer_offset(const pointer_offset_exprt &expr)
       return unchanged(expr);
 
     // this might change the type of the pointer!
-    exprt pointer_offset_expr = simplify_pointer_offset(
-      to_pointer_offset_expr(pointer_offset(ptr_expr.front())));
+    exprt pointer_offset_expr = simplify_pointer_offset(to_pointer_offset_expr(
+      pointer_offset_exprt(ptr_expr.front(), expr.type())));
```

Comment:
I don't understand this change: why do we do `to_pointer_offset_expr` when the argument (now) is a `pointer_offset_exprt`? And when we're now constructing the expression with `expr.type()`, does the comment still apply? If the comment no longer applies, why was it previously ok to change the type, and why is it now the right thing not to change the type?

Reply:
1. Removed the comment, this is indeed no longer the case.
2. Remove the cast.